sw/device/tests/penetrationtests/firmware/ibex_fi.c

```diff
@@ -547,26 +568,32 @@ status_t handle_ibex_fi_char_sram_read(ujson_t *uj) {
       mmio_region_from_addr(sram_main_buffer_addr);
 
   // Write reference values into SRAM.
-  for (int i = 0; i < 32; i++) {
+  for (int i = 0; i < 256; i++) {
     mmio_region_write32(sram_region_main_addr, i * (ptrdiff_t)sizeof(uint32_t),
-                        ref_values[i]);
+                        ref_values[i % 32]);
```

Comment:
it may be helpful to not mod 32 to be able to distinguish the 256 values, but on the other hand, if address is corrupt it will read other initialized mem with zeros (?) which we cannot dinstinguish anyways

Reply:
Yeah exactly, reading from a faulty address would, with a high probability, return something from memory.